paddle/fluid/framework/fleet/box_wrapper.h

```diff
@@ -413,6 +413,42 @@ class BoxWrapper {
     std::vector<std::pair<int, int>> cmatch_rank_v;
     std::string cmatch_rank_varname_;
   };
+  class MaskMetricMsg : public MetricMsg {
+   public:
+    MaskMetricMsg(const std::string& label_varname,
+                  const std::string& pred_varname, int is_join,
+                  const std::string& mask_varname, int bucket_size = 1000000) {
+      label_varname_ = label_varname;
+      pred_varname_ = pred_varname;
+      mask_varname_ = mask_varname;
+      is_join_ = is_join;
+      calculator = new BasicAucCalculator();
+      calculator->init(bucket_size);
+    }
+    virtual ~MaskMetricMsg() {}
+    void add_data(const Scope* exe_scope) override {
+      std::vector<int64_t> label_data;
+      get_data<int64_t>(exe_scope, label_varname_, &label_data);
+      std::vector<float> pred_data;
+      get_data<float>(exe_scope, pred_varname_, &pred_data);
+      std::vector<int64_t> mask_data;
+      get_data<int64_t>(exe_scope, mask_varname_, &mask_data);
+      auto cal = GetCalculator();
+      auto batch_size = label_data.size();
+      std::string out = "";
+      for (size_t i = 0; i < batch_size; ++i) {
+        out += std::to_string(mask_data[i]) + " ";
```

Comment:
what dose variable 'out' mean?

Reply:
used for testing. Deleted it done. Thanks!